tests/RobotFramework/tests/tedge_agent/device_profile/device_profile.toml

```diff
@@ -0,0 +1,69 @@
+operation = "device_profile"
+
+[init]
+action = "proceed"
+on_success = "scheduled"
+
+[scheduled]
+script = "/etc/tedge/operations/device_profile.sh ${.payload.status} ${.payload}"
+on_success = "next_artifact"
+
+[next_artifact]
+script = "/etc/tedge/operations/device_profile.sh ${.payload.status} ${.payload}"
+on_stdout = ["firmware_update", "software_update", "config_update", "successful", "failed"]
+on_error = { status = "failed", reason = "fail to run the script"}
+
+#
+# Firmware
+#
+[firmware_update]
+operation = "firmware_update"
+input.name = "${.payload.current.payload.name}"
+input.version = "${.payload.current.payload.version}"
+input.remoteUrl = "${.payload.current.payload.remoteUrl}"
+on_exec = "waiting_for_firmware_update"
+
+[waiting_for_firmware_update]
+action = "await-operation-completion"
+on_success = "next_artifact"
+on_error = { status = "failed", reason = "fail to update the firmware"}
+
+#
+# Software
+#
+[software_update]
+operation = "software_update"
+input.updateList = "${.payload.current.payload.updateList}"
+on_exec = "waiting_for_software_update"
+
+[waiting_for_software_update]
+action = "await-operation-completion"
+on_success = "next_artifact"
+on_error = { status = "failed", reason = "fail to install the software"}
+
+#
+# Configuration
+#
+[config_update]
+operation = "config_update"
+input.type = "${.payload.current.payload.type}"
+input.remoteUrl = "${.payload.current.payload.remoteUrl}"
+on_exec = "waiting_for_config_update"
+
+[waiting_for_config_update]
+action = "await-operation-completion"
+on_success = "next_artifact"
+on_error = { status = "failed", reason = "fail to update configuration"}
+
+#
+# End states
+#
+[successful]
+action = "cleanup"
+
+[failed]
+action = "cleanup"
+
+# Points
+# * How to do conditional actions, e.g. only try installing firmware if there is firmware to be installed
+# * Allow variable expansion on the "operation" state property
```

Comment:
This can easily be added. Will do in https://github.com/thin-edge/thin-edge.io/pull/2918

Reply:
Done: https://github.com/thin-edge/thin-edge.io/pull/2918/commits/81d22b3c41c627488006545f7a2abb577fea2e45